packages/std/src/traits.rs

```diff
@@ -102,13 +111,13 @@ pub trait Querier: Clone + Send {
             address: address.into(),
             denom: denom.to_string(),
         });
-        self.parse_query(&request)
+        self.parse_query::<NoMsg, _>(&request)
```

Comment:
This gives us `QueryRequest::Custom(NoMsg)`, so we better generalize the type to `Never`

Reply:
If you like that, I can make a little service PR to get that changed independent of the query topic.

Reply:
I don't understand.
This will parse all non-custom QueryRequest entries just fine. And as we are not using custom, we can hardcode this.

Do you mean to rename NoMsg to Never?

Reply:
Yes, the point is renaming because it was created for a specific use case (messages) and passing it as an argument for query requests is confusing.

Reply:
Yeah. I would actually do something else.

Make query only take one generic, which is the return value and always use NoMsg under the hood. This is fine for any non custom query.

Then add a query_custom method that takes the second argument and specifies the custom type. This is equivalent to the current parse_query method, but only needed when we are using the Custom type.

Wdyt?

Reply:
It still uses the type, but hides it.

We can also rename it to Never, but this is about removing it from the API

Reply:
Wouldn't this approach conflict with the current idea of `query_balance`/`query_all_balances`/... for commonly used queries?

Reply:
Renamed to `Never` in 32377c9